### ISSUE TITLE
implement sign in with github

### ISSUE THREAD

**Issue by vj-codes** [anitab-org/mentorship-backend#1119]
## Description

As a developer,
I need to add login with GitHub feature to app,
so that I can allow users to use their GitHub OAuth.

Implement login with GitHub to mentorship system

https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps

### Update [Required]
- [ ] A user is able to login into the platform with GitHub OAuth

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation

???

**Comment by vj-codes** [comment]
I would like to work on this

**Comment by epicadk** [comment]
On my phone so can't assign for some reason. Please self assign.

**Comment by vj-codes** [comment]
> On my phone so can't assign for some reason. Please self assign.

afaik the github app doesn't include that feature yet, you would need to use the browser on phone to do the same

**Comment by vj-codes** [comment]
@isabelcosta @epicadk for this there will be a need to create an OAuth app on Github that will give me client secret and client id which I can create for now from my acc but later the org acc will need to do the same
Is that alright?

**Comment by vj-codes** [comment]
Update : was kinda busy with exams still working on it, got a few questions will ask in the sync up session if I don't find a solution about them on StackOverflow before the session 🚀

**Comment by devkapilbansal** [comment]
@vj-codes any updates?

**Comment by vj-codes** [comment]
Oops I forgot about this 😄 
Making it available as I can't work on it right now :)

**Comment by isabelcosta** [comment]
Thank you @vj-codes for updating the issue 🙌🏾

**Comment by lem0n4id** [comment]
Hey, i would like to work on this.

Can anyone please **list out** what all needs to be done to *Implement login with GitHub to mentorship system*?
I am new here

**Comment by lem0n4id** [comment]
hello, sorry for pinging everyone in this conversation.
i will not be able to work on this as i am not familiar with flask

**Comment by isabelcosta** [comment]
No worries @lem0n4id in case it helps:
- We are using Flask Restplus - https://flask-restplus.readthedocs.io/en/stable/
- Our documentation for this project: https://anitab-org.github.io/mentorship-backend/docs/

Feel also free to look other issues that can ease your way into the app ;)

**Comment by sukesh2000** [comment]
Is this issue still open, can I work on this? I've experience with implementing OAuth2 in NodeJS & Flask.

**Comment by devkapilbansal** [comment]
@sukesh2000 yes you can work on this. I have assigned this issue to you. Feel free to move forward and ask here or on Zulip if you have some doubts. 
Happy Coding 😃

**Comment by sukesh2000** [comment]
Thank you so much, I'll start working on it! I've joined Zulip as well, if I've any doubts then I'll put there.

**Comment by vj-codes** [comment]
Any updates @sukesh2000 ?
